x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportPutDatafeedAction.java

```diff
@@ -124,33 +136,74 @@ private void handlePrivsResponse(String username, PutDatafeedAction.Request requ
     private void putDatafeed(PutDatafeedAction.Request request, Map<String, String> headers,
                              ActionListener<PutDatafeedAction.Response> listener) {
 
-        clusterService.submitStateUpdateTask(
-                "put-datafeed-" + request.getDatafeed().getId(),
-                new AckedClusterStateUpdateTask<PutDatafeedAction.Response>(request, listener) {
+        String datafeedId = request.getDatafeed().getId();
+        String jobId = request.getDatafeed().getJobId();
+        ElasticsearchException validationError = checkConfigsAreNotDefinedInClusterState(datafeedId, jobId);
```

Comment:
Protection against the datefeed or job being defined in the clusterstate. We may not need this in the long run but it's prudent to have now for detecting bugs.